bundles/ch.elexis.core.ui/src/ch/elexis/core/ui/views/rechnung/RnOutputDialog.java

```diff
@@ -66,39 +78,48 @@ protected Control createDialogArea(Composite parent) {
 			SWTHelper.alert(msg, msg);
 			return null;
 		}
-		Composite ret = new Composite(parent, SWT.NONE);
+		sortOutputters(lo);
+		ret = new Composite(parent, SWT.NONE);
 		ret.setLayout(new GridLayout());
 		ret.setLayoutData(SWTHelper.getFillGridData(1, true, 1, true));
+		GridData gridData = new GridData(SWT.FILL, SWT.CENTER, true, false);
+		gridData.horizontalIndent = 5;
 		cbLo = new Combo(ret, SWT.SINGLE | SWT.READ_ONLY);
-		cbLo.setLayoutData(SWTHelper.getFillGridData(1, true, 1, false));
+		cbLo.setLayoutData(gridData);
 		bCopy = new Button(ret, SWT.CHECK);
-		bCopy.setText(Messages.RnOutputDialog_markAsCopy); // $NON-NLS-1$
-		bCopy.setLayoutData(SWTHelper.getFillGridData(1, true, 1, false));
+		bCopy.setText(Messages.RnOutputDialog_markAsCopy);
+
+		bCopy.setLayoutData(gridData);
 		final Composite bottom = new Composite(ret, SWT.NONE);
 		bottom.setLayoutData(SWTHelper.getFillGridData(1, true, 1, true));
 		bottom.setLayout(stack);
 		for (IRnOutputter ro : lo) {
 			cbLo.add(ro.getDescription());
-			ctls.add((Control) ro.createSettingsControl(bottom));
+			Control control = (Control) ro.createSettingsControl(bottom);
+			GridData controlGridData = new GridData();
+			controlGridData.horizontalIndent = 50;
+			control.setLayoutData(controlGridData);
+			ctls.add(control);
 		}
 		cbLo.addSelectionListener(new SelectionAdapter() {
 			@Override
 			public void widgetSelected(SelectionEvent e) {
 				int idx = cbLo.getSelectionIndex();
 				if (idx != -1) {
+					IRnOutputter selectedOutputter = lo.get(idx);
 					customizeDialog(lo.get(idx));
 					stack.topControl = ctls.get(idx);
 					bottom.layout();
-					CoreHub.localCfg.set(Preferences.RNN_DEFAULTEXPORTMODE, idx);
+					LocalConfigService.set(Preferences.RNN_DEFAULTEXPORTMODE, idx);
+					if (isOkButtonEnabledFor(selectedOutputter)) {
+						setOkButtonEnabled(true);
+					}
```

Comment:
Weshalb wir das hier gemacht?

Die Methode `customizeDialog` wird hier auf Zeile 110 aufgerufen, und ruft dann beim selektierten `IRnOutputter#customizeDialog` auf.

In https://github.com/elexis/elexis-3-base/pull/336 wird Z.B in [QrRnOutputter.java](https://github.com/elexis/elexis-3-base/pull/336/files#diff-cc9ea7237246727b999e85a5a58ff0554969c1cad4fb6707c72b12cbc71862d8) die `customizeDialog` Methode überschrieben, und darin die Methode `updateButtonStates` aufgerufen. Dort wird dann `setOkButtonEnabled` aufgerufen.

D.h. es könnte in `QrRnOutputter` auch die Logik angepasst werden das der OK wie erwartet enabled ist.